plugin/evm/vm.go

```diff
@@ -351,6 +362,10 @@ func (vm *VM) Initialize(
 	vm.chain = ethChain
 	lastAccepted := vm.chain.LastAcceptedBlock()
 
+	if err := vm.initializeMetrics(); err != nil {
+		return err
+	}
+
 	// start goroutines to update the tx pool gas minimum gas price when upgrades go into effect
 	vm.handleGasPriceUpdates()
```

Comment:
in coreth, this happens after the peer network is created:
* https://github.com/ava-labs/coreth-internal/blob/9fb7a2025a20f3a5546683571f1066cae932457c/plugin/evm/vm.go#L473
* https://github.com/ava-labs/coreth-internal/blob/9fb7a2025a20f3a5546683571f1066cae932457c/plugin/evm/vm.go#L546